client/extensions/zoninator/state/data-layer/zones/index.js

```diff
@@ -24,8 +33,50 @@ export const requestZonesError = ( { dispatch }, { siteId } ) =>
 export const updateZonesList = ( { dispatch }, { siteId }, { data } ) =>
 	dispatch( updateZones( siteId, data ) );
 
+export const createZone = ( { dispatch }, action ) => {
+	const { data, form, siteId } = action;
+
+	dispatch( startSubmit( form ) );
+	dispatch( removeNotice( 'zoninator-zone-create' ) );
+	dispatch( http( {
+		method: 'POST',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			body: JSON.stringify( data ),
+			json: true,
+			path: '/zoninator/v1/zones',
+		},
+	}, action ) );
+};
+
+export const announceZoneSaved = ( { dispatch }, { form, siteId }, next, { data } ) => {
+	dispatch( stopSubmit( form ) );
+	dispatch( updateZone( siteId, data ) );
+	dispatch( successNotice(
+		translate( 'Zone saved!' ),
+		{ id: 'zoninator-zone-create' },
+	) );
+};
+
+export const announceZoneCreated = ( { dispatch, getState }, action, next, response ) => {
+	const { siteId } = action;
+
+	page( `/extensions/zoninator/${ getSiteSlug( getState(), siteId ) }` );
+	announceZoneSaved( { dispatch }, action, next, response );
```

Comment:
The idea here was that `announceZoneSaved` could be re-used for the upper section of 'Edit Zone' form ( #16497 ), as it does pretty much the same thing minus the redirect.

The two sections on `Edit Zone` page will actually be two separate forms as they're handled by two separate endpoints.